modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm

```diff
@@ -0,0 +1,13 @@
+/**
+ * Checks if a player meets certain conditions to exclude them from event selection.
+ */
+/proc/engaged_role_play_check(mob/living/carbon/human/player, station = TRUE, dorms = TRUE)
+	var/turf/player_turf = get_turf(player)
+	var/area/player_area = get_area(player_turf)
+	if(!is_station_level(player_turf.z) && station)
+		return TRUE
+
+	if(istype(player_area, /area/station/commons/dorms) && dorms)
```

Comment:
In the face of my previous suggestion, the deep space hotel should ward off those random events.

Reply:
Right now, yes, but should LT3 change the off-station check to an on-interlink / ghost cafe check that won't be longer the case, Goof.

Reply:
We can add more granularity, but I don't want disease outbreak wasting itself anywhere off station.